GhidraJupyterKotlin/src/main/java/GhidraJupyterKotlin/extensions/data/DataExtensions.kt

```diff
@@ -0,0 +1,24 @@
+package GhidraJupyterKotlin.extensions.data
+
+import ghidra.program.model.data.Structure
+import ghidra.program.model.listing.Data
+
+
+
+// For a Data object that supports component (like arrays or structs) you can use
+// `data[i]` instead of `data.getComponent(i)`
+operator fun Data.get(i: Int): Data? {
+    return this.getComponent(i)
+}
+
+// For a Data object that represents a struct you can use
+// `data[fieldName]`
+
+operator fun Data.get(name: String): Data? {
```

Comment:
I wonder if it's not better to add this as an extension for `Structure`, and always require a cast. 
But that's probably just making it harder to use for no real benefit.

Reply:
I don't think that works at all, `getDataAt(someAddress).dataType` is of type `Structure` (or whatever else for other kinds of data). the result of getDataAt() is always just `Data`. DataTypes like `Structure` also have a method for getComponent but that only returns the struct member and you can't get back to the actual value in memory because that information is in the `Data` object

Reply:
the actual implementation of Data.getComponent has a bunch of branches for `baseDataType instanceof Composite`, `baseDataType instanceof Structure` etc as well...